torchao/prototype/mx_formats/mx_linear.py

```diff
@@ -15,21 +15,41 @@
 from torch.distributed._tensor import DTensor
 
 from torchao.prototype.mx_formats.config import (
+    MXFP8CastKernelChoice,
     MXGemmKernelChoice,
     MXInferenceLinearConfig,
     MXLinearConfig,
 )
-from torchao.prototype.mx_formats.kernels import triton_to_mxfp8_dim1
+from torchao.prototype.mx_formats.kernels import (
+    mxfp8_quantize_cuda,
+    triton_to_mxfp8_dim1,
+)
 from torchao.prototype.mx_formats.mx_tensor import MXTensor
 from torchao.quantization.transform_module import (
     register_quantize_module_handler,
 )
 
 
-def _triton_to_mxfp8_dim1_wrapper(
-    a, block_size, elem_dtype, hp_dtype, gemm_kernel_choice
+def _to_mxfp8_dim1_kernel_wrapper(
+    a,
+    block_size,
+    elem_dtype,
+    hp_dtype,
+    gemm_kernel_choice,
+    cast_kernel_choice,
 ):
-    a_data, a_scale = triton_to_mxfp8_dim1(a, block_size)
+    if cast_kernel_choice == MXFP8CastKernelChoice.TRITON:
+        a_data, a_scale = triton_to_mxfp8_dim1(a, block_size)
+    elif cast_kernel_choice == MXFP8CastKernelChoice.CUDA:
+        _, a_data, _, a_scale = mxfp8_quantize_cuda(
+            a,
+            rowwise=False,
+            colwise=True,
+            scaling_mode="floor",
```

Comment:
TODO for later to allow choice of scaling modes

Reply:
Added todo on the custom op itself, with an explanation why we currently are using a string param